asdf/extension/_manager.py

```diff
@@ -214,3 +225,80 @@ def get_cached_extension_manager(extensions):
 @lru_cache
 def _get_cached_extension_manager(extensions):
     return ExtensionManager(extensions)
+
+
+class ValidatorManager:
+    """
+    Wraps a list of validators and indexes them by schema property.
+
+    Parameters
+    ----------
+    validators : iterable of asdf.extension.Validator
+        List of validators to manage.
+    """
+
+    def __init__(self, validators):
+        self._validators = list(validators)
+
+        self._validators_by_schema_property = {}
+        for validator in self._validators:
+            if validator.schema_property not in self._validators_by_schema_property:
+                self._validators_by_schema_property[validator.schema_property] = set()
+            self._validators_by_schema_property[validator.schema_property].add(validator)
+
+    def validate(self, schema_property, schema_property_value, node, schema):
+        """
+        Validate an ASDF tree node against a schema property.
+
+        Parameters
+        ----------
+        schema_property : str
+            Name of the schema property (identifies the validator(s) to use).
+        schema_property_value : object
+            Value of the schema property.
+        node : asdf.tagged.Tagged
+            The ASDF node to validate.
+        schema : dict
+            The schema object that contains the property that triggered
+            the validation.
+
+        Yields
+        ------
+        asdf.exceptions.ValidationError
+        """
+        if schema_property in self._validators_by_schema_property:
+            for validator in self._validators_by_schema_property[schema_property]:
+                if _validator_matches(validator, node):
+                    yield from validator.validate(schema_property_value, node, schema)
+
+    def get_jsonschema_validators(self):
+        """
+        Get a dictionary of validator methods suitable for use
+        with the jsonschema library.
+
+        Returns
+        -------
+        dict of str: callable
+        """
+        result = {}
+
+        for schema_property in self._validators_by_schema_property:
+            result[schema_property] = self._get_jsonschema_validator(schema_property)
+
+        return result
+
+    def _get_jsonschema_validator(self, schema_property):
+        def _validator(_, schema_property_value, node, schema):
+            return self.validate(schema_property, schema_property_value, node, schema)
+
+        return _validator
+
+
+def _validator_matches(validator, node):
+    if any(t == "**" for t in validator.tags):
+        return True
+
+    if not isinstance(node, Tagged):
+        return False
```

Comment:
We should add a test to cover these cases since the numpy validators no longer test them.

Reply:
I opened a PR on your branch that should add coverage to these lines:
https://github.com/eslavich/asdf/pull/3